### PR TITLE
wire up apd overview front-end

### PR DESCRIPTION
**Pull request by brendansudol**
Allows user to add/edit/save their program vision and benefits.

### This pull request is ready to merge when...
- [ ] Tests have been updated (and all tests are passing)
- [ ] This code has been reviewed by someone other than the original author
- [ ] The experience passes a basic manual accessibility audit (keyboard nav, screenreader, text scaling) OR an exemption is documented
- [ ] The change has been documented
  - [ ] Associated OpenAPI documentation has been updated

### CODE REVIEW

File: api/routes/states/put.js [hunk 1]
```diff
@@ -64,7 +64,12 @@ module.exports = (app, dataHelper = defaultDataHelper) => {
       logger.silly(req, newData);
 
       logger.silly(req, 'validating program benefits');
-      if (joi.string().validate(newData.program_benefits).error) {
+      if (
+        joi
+          .string()
+          .allow('')
```
Comment (brendansudol) [inline api/routes/states/put.js]: empty string was failing validation, and we want to accept it if user wants to clear out their prior input.

https://github.com/hapijs/joi/issues/448

**Comment by brendansudol** [comment]
This should close: https://github.com/18F/cms-hitech-apd/issues/279

File: api/routes/states/put.js (hunk 1 above)
Comment (mgwalker) [reply api/routes/states/put.js]: Interesting!

File: web/src/components/FormApdOverview.js [hunk 2]
```diff
@@ -5,27 +5,38 @@ import { Field, reduxForm } from 'redux-form';
...
 FormApdOverview.propTypes = {
-  handleSubmit: PropTypes.func.isRequired
+  handleSubmit: PropTypes.func.isRequired,
+  submitting: PropTypes.bool.isRequired
 };
 
 const formConfig = {
   form: 'apdOverview',
-  destroyOnUnmount: false
+  destroyOnUnmount: false,
```
Comment (mgwalker) [inline web/src/components/FormApdOverview.js]: I wonder if we can get rid of the `destroyOnUnmount` for the forms that are wired up now.  We take the submitted data, send it to the API, then update the app state with the returned data.  That is then used to set the `initialValues`, so if the form is destroyed when we move on to the next section, it's no big deal - it gets rebuilt with the data the API sent back after saving previously.  Is that right?  I'm not sure if it's a big deal, but I assume there's a reason it defaults to destroying the forms. 🤷‍♂️

File: web/src/containers/ApdOverview.js [hunk 3]
```diff
@@ -1,42 +1,79 @@
...
 ApdOverview.propTypes = {
-  goTo: PropTypes.func.isRequired
+  goTo: PropTypes.func.isRequired,
+  fetchStateDataIfNeeded: PropTypes.func.isRequired,
+  updateState: PropTypes.func.isRequired,
+  formData: PropTypes.object.isRequired,
+  stateInfo: PropTypes.object.isRequired
+};
+
+const mapStateToProps = ({ state }) => {
```
Comment (mgwalker) [inline web/src/containers/ApdOverview.js]: This is our state state.  👍

**Pull request merged**